crates/bevy_animation/src/lib.rs

```diff
@@ -138,68 +150,92 @@ pub enum Interpolation {
     CubicSpline,
 }
 
-/// Path to an entity, with [`Name`]s. Each entity in a path must have a name.
-#[derive(Reflect, Clone, Debug, Hash, PartialEq, Eq, Default)]
-pub struct EntityPath {
-    /// Parts of the path
-    pub parts: Vec<Name>,
-}
-
-/// A list of [`VariableCurve`], and the [`EntityPath`] to which they apply.
+/// A list of [`VariableCurve`]s and the [`AnimationTargetId`]s to which they
+/// apply.
+///
+/// Because animation clips refer to targets by UUID, they can target any
+/// [`AnimationTarget`] with that ID.
 #[derive(Asset, Reflect, Clone, Debug, Default)]
 pub struct AnimationClip {
-    curves: Vec<Vec<VariableCurve>>,
-    paths: HashMap<EntityPath, usize>,
+    curves: AnimationCurves,
     duration: f32,
 }
 
+/// A mapping from [`AnimationTargetId`] (bone) to the animation curves.
+pub type AnimationCurves = HashMap<AnimationTargetId, Vec<VariableCurve>, NoOpHash>;
+
+/// A unique [UUID] for an animation target (bone).
+///
+/// The [`AnimationClip`] asset and the [`AnimationTarget`] component both use
+/// this to refer to targets (bones) to be animated.
+///
+/// When importing an armature or an animation clip, asset loaders typically use
+/// the full path name from the armature to the bone to generate these IDs. So,
+/// for example, any imported armature with a bone at the root named `Hips` will
+/// assign the same [`AnimationTargetId`] to its root bone. Likewise, any
+/// imported animation clip that animates a root bone named `Hips` will
+/// reference the same [`AnimationTargetId`]. What this means is that any
+/// animation is playable on any armature as long as the bone names match,
+/// which allows for easy animation retargeting.
+///
+/// Note that asset loaders generally use the *full* path name to generate the
+/// [`AnimationTargetId`]. Thus a bone named `Chest` directly connected to a
+/// bone named `Hips` will have a different ID from a bone named `Chest` that's
+/// connected to a bone named `Stomach`.
+///
+/// [UUID]: https://en.wikipedia.org/wiki/Universally_unique_identifier
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Reflect, Debug)]
+pub struct AnimationTargetId(pub Uuid);
```

Comment:
This would break asset hot reload, prevent animation sharing, and would mean games with a lot of animations, especially heavy ones for cinematics, would need to be loaded all the time. Probably not ideal.

Reply:
@djeedai Just to clarify, you're talking about @james7132's proposal, not this PR, right? I'm pretty sure this PR doesn't break either of those things.

Anyway, as I was planning to implement it I think that it's possible to have hot asset reload without reconstructing the graph. You'd just have to invalidate the `tracks` and rebuild it on the fly, treating `tracks` as a cache rather than the main source of truth. It's the perpetual tradeoff of caching, unfortunately: if you're in a situation in which you're trashing the cache, the cache actually loses...

Reply:
Perhaps this deserves a separate PR? I think it is probably worth it, but I'd like time to evaluate it.

Reply:
Yeah, I can leave it out of the first `AnimationGraph` PR. Anyway, this is all not relevant to this PR; probably better to discuss on Discord.